_rules/widget-non-empty-accessible-name-rdzs6q.md

```diff
@@ -0,0 +1,382 @@
+
+- `button`; or
+- `checkbox`; or
+- `combobox`; or
+- `link`; or
+- `listbox`; or
+- `menuitem`; or
+- `menuitemcheckbox`; or
+- `menuitemradio`; or
+- `radio`; or
+- `searchbox`; or
+- `slider`; or
+- `spinbutton`; or
+- `switch`; or
+- `textbox`.
+
+## Expectation
+
+Each target element has an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+- Certain assistive technologies can be set up to ignore the title attribute, which means that to some users the title attribute will not act as an [accessible name][].
+- Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have one of the applicable [semantic roles][] and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+- Accessibility support for some elements inheriting the semantic role of `link` (e.g. elements with [`doc-*` roles](https://www.w3.org/TR/dpub-aria-1.1/#role_definitions)) may vary depending on the assistive technology in use.
+- Some elements have a role of `button` and a default accessible name defined by the [HTML Accessibility API Mapping][html aam input button], for example `input` elements whose `type` [attribute value][] is either `submit` or `reset`. This rule considers that these default names can be descriptive and therefore does not fail them.
+
+## Background
+
+The list of roles in the applicability considers elements that meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). It comprises the `button`, `link` and `menuitem` roles, plus all the roles from [WAI-ARIA Specifications](#wai-aria-specifications) that:
+
+- have [semantic roles][] that inherit from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and
+- do not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+
+Elements with the `option` role are not tested in this rule because they do not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). If these elements are presented as user interface components, these need to be tested separately from this rule.
+
+Unlike the [Image button has non-empty accessible name](https://act-rules.github.io/rules/59796f) rule, this rule does not check whether an element's accessible name is different from the default name. While that rule assumes that, for image buttons, the default name is never descriptive, this rule relaxes that assumption to ensure the simplicity of this rule.
+
+### Related rules
+
+- [Image button has non-empty accessible name](https://act-rules.github.io/rules/59796f)
+- [Link has non-empty accessible name](https://act-rules.github.io/rules/c487ae)
```

Comment:
if this c487ae rule isn't getting deprecated, should link be removed from this widget rule? Otherwise, we'll have 2 rules to test link has non-empty accessible name.